Sources/Host/POSIX/Platform.cpp

```diff
@@ -124,6 +127,23 @@ bool Platform::SetWorkingDirectory(std::string const &directory) {
 
 ds2::ProcessId Platform::GetCurrentProcessId() { return ::getpid(); }
 
+// Terminates the process and waits for it to exit before returning.
+bool Platform::TerminateProcess(ProcessId pid) {
+  if (::kill(pid, SIGKILL) < 0 && errno != ESRCH)
```

Comment:
if errno is ESRCH would it make sense to return true right away? since there's a chance there could be a race condition until waitpid where the system reuses the pid and launches a different process.

Reply:
and true kinda makes sense since the process is already terminated

Reply:
On Linux, the pid remains valid (e.g. zombie process) and won't be reused until the parent waits on it. I don't know if that's the case for other posix systems, though, and since I did implement this method as a generic posix solution... I'll go check on that.

Reply:
A cursory search suggests this behavior should be the same on FreeBSD and MacOS (and posix in general). Let me know if that's not your understanding.

Reply:
ok, sounds good!